projects/kit/components/slider/slider-key-steps.directive.ts

```diff
@@ -0,0 +1,117 @@
+import {
+    ChangeDetectorRef,
+    Directive,
+    ElementRef,
+    forwardRef,
+    HostListener,
+    Inject,
+    Input,
+    Optional,
+    Self,
+} from '@angular/core';
+import {NgControl} from '@angular/forms';
+import {
+    AbstractTuiControl,
+    isNativeFocused,
+    TUI_FOCUSABLE_ITEM_ACCESSOR,
+    tuiDefaultProp,
+    TuiFocusableElementAccessor,
+} from '@taiga-ui/cdk';
+import {TuiKeySteps} from '@taiga-ui/kit/types';
+
+import {TuiSliderComponent} from './slider.component';
+
+@Directive({
+    selector: 'input[tuiSlider][keySteps]',
+    providers: [
+        {
+            provide: TUI_FOCUSABLE_ITEM_ACCESSOR,
+            useExisting: forwardRef(() => TuiSliderComponent),
+        },
+    ],
+})
+export class TuiSliderKeyStepsDirective
+    extends AbstractTuiControl<number>
+    implements TuiFocusableElementAccessor
+{
+    @Input()
+    @tuiDefaultProp(
+        (steps: TuiKeySteps) =>
+            !steps.length || (steps[0][0] === 0 && steps[steps.length - 1][0] === 100),
```

Comment:
`steps?.[0]?.[0]`, `steps[steps.length - 1]?.[0]` not sure about it, but maybe is good for type safety

Reply:
`!steps.length` ensures that we get array with at least 1 item.
Typescript ensures that this item contains two numbers inside.

I am not sure that we need optional chaining here